product-scenarios/scenarios-commons/src/main/java/org/wso2/carbon/esb/scenario/test/common/vfs/VFSClient.java

```
@@ -0,0 +1,222 @@
+     * @param targetFilePath target path of the remote file
+     * @throws IOException if file upload fails
+     */
+    public void ftpFileUpload(String sourceFilePath, String targetFilePath) throws IOException {
+        ftpClient.setFileType(FTP.BINARY_FILE_TYPE);
+        File localFile = new File(sourceFilePath);
+
+        try (InputStream inputStream = new FileInputStream(localFile)) {
+            log.info("Start uploading file " + sourceFilePath + " to " + targetFilePath);
+
+            try {
+                if (ftpClient.storeFile(targetFilePath, inputStream)) {
+                    log.info("File has been uploaded to " + targetFilePath + " successfully.");
+                }
+            } catch (IOException ex) {
+                throw new IOException("IO Exception occurred while uploading file ", ex);
+            }
+        } catch (FileNotFoundException ex) {
+            throw new IOException("File Not Found Exception occurred ", ex);
+        }
+    }
+
+    /**
+     * Function to download a file from an FTP Server
+     *
+     * @param sourceFilePath source file path
+     * @param targetFilePath target path
+     * @throws IOException if file download fails
+     */
+    public void ftpFileDownload(String sourceFilePath, File targetFilePath) throws IOException{
+        ftpClient.setFileType(FTP.BINARY_FILE_TYPE);
+
+        try (OutputStream outputStream1 = new BufferedOutputStream(new FileOutputStream(targetFilePath))) {
+            log.info("Start downloading file " + sourceFilePath + " to " + targetFilePath);
+
+            try {
+                if (ftpClient.retrieveFile(sourceFilePath, outputStream1)) {
+                    log.info("File has been uploaded to " + targetFilePath + " successfully.");
+                }
+            } catch (IOException ex) {
+                throw new IOException("IO Exception occurred while downloading file ", ex);
+            }
+        } catch (FileNotFoundException ex) {
```

Let's just add the exception to the method signature. I suppose we do not gain anything by catching it and creating a new exception. Shall we correct in all places?

---

In this function IO Exception would throw when downloading the file fails or unable to find the file. So isn't it better to throw the exception with a descriptive message?